gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -672,15 +735,16 @@ function roleselection.SelectRoles(plys, maxPlys)
 	local selectableRoles = roleselection.GetSelectableRolesList(maxPlys, allAvailableRoles) -- update roleselection.selectableRoles table
 
 	-- Select forced roles at first
-	local selectedForcedPlys = SelectForcedRoles(plys, selectableRoles) -- this updates roleselection.finalRoles table and returns a key based list of selected players
+	local selectedForcedRoles = GetHardForcedBaseRoles() -- this gets the amount of baseroles that are taken by subroles 
+	SelectForcedRoles(plys, selectableRoles, selectedForcedRoles) -- this updates roleselection.finalRoles table and the list of forced Roles
 
 	-- We need to remove already selected players
 	local plysFirstPass, plysSecondPass = {}, {} -- modified player table
 
 	for i = 1, #plys do
 		local ply = plys[i]
 
-		if selectedForcedPlys[ply] then continue end
+		if roleselection.finalRoles[ply] then continue end
```

Comment:
This seemed kinda strange to remove already selected Players, but only check for selectedForcedPlys, therefore I changed that to the finalRoles List, which should contain both.

Reply:
So you don't need this var localized here anymore? https://github.com/TTT-2/TTT2/pull/685/files#diff-d2698bbe621f7660fa9806073e9a4fe3e10977cb1c48a345babc478d6f3e44a9R736

Reply:
Well I not for the here marked part, but for this input https://github.com/TTT-2/TTT2/pull/685/files#diff-d2698bbe621f7660fa9806073e9a4fe3e10977cb1c48a345babc478d6f3e44a9R737

Or should it be put into the function itself or instead of a parameter use the function.
Depends on the style you like more. I'd prefer it localized there to make it clear what its used for.

Reply:
I think it would match the best directly inside the SelectForcedRoles fn